app/path.go

```diff
@@ -228,18 +231,49 @@ func collectSchemaNodes(e *yang.Entry, leafOnly bool) []*yang.Entry {
 		collected = append(collected,
 			collectSchemaNodes(child, leafOnly)...)
 	}
+
 	if e.Parent != nil {
 		switch {
 		case e.Dir == nil && e.ListAttr != nil: // leaf-list
 			fallthrough
 		case e.Dir == nil: // leaf
-			collected = append(collected, e)
+			f := &yang.Entry{
+				Parent:      e.Parent,
+				Node:        e.Node,
+				Name:        e.Name,
+				Description: e.Description,
+				Default:     e.Default,
+				Units:       e.Units,
+				Kind:        e.Kind,
+				Config:      e.Config,
+				Prefix:      e.Prefix,
+				Mandatory:   e.Mandatory,
+				Dir:         e.Dir,
+				Key:         e.Key,
+				Type:        e.Type,
+				Exts:        e.Exts,
+				ListAttr:    e.ListAttr,
+				Extra:       make(map[string][]interface{}),
```

Comment:
given `any` is introduced in go1.18 for an empty interface it might be good to introduce new code with it